tests/testAfterburners.py

```diff
@@ -0,0 +1,181 @@
+        measRecord.set(self.failKey, True)
+
+@register("singleRecordAfterburner")
+class SingleRecordAb(AB.AfterburnerPlugin):
+    '''
+    Afterburner plugin which works in single mode. It takes a single record, reads a value, squares it, and
+    writes out the results to the record
+    '''
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_AFTERBURNER
+
+    def __init__(self, config, name, schema, metadata):
+        AB.AfterburnerPlugin.__init__(self, config, name, schema, metadata)
+        self.failKey = schema.addField(name+"_fail", type="Flag", doc="Failure flag")
+        self.squareKey = schema.addField(name+"_square", type="D", doc="Square of input catalog")
+
+    def burn(self, measRecord):
+        value = measRecord.get("start")
+        measRecord.set(self.squareKey, value**2)
+
+    def fail(self, measRecord, error=None):
+        measRecord.set(self.failKey, True)
+
+@register("multiRecordAfterburner")
+class MultiRecordAb(AB.AfterburnerPlugin):
+    '''
+    Afterburner plugin to test the framework in multimode. This plugin takes the whole source catalog at
+    onece, and loops over the catalog internally. The algorithm simply reads a value, cubes it, and writes the
+    results out to the table
+    '''
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_AFTERBURNER
+
+    @property
+    def PlugType(self):
+        return 'multi'
+
+    def __init__(self, config, name, schema, metadata):
+        AB.AfterburnerPlugin.__init__(self, config, name, schema, metadata)
+        self.failKey = schema.addField(name+"_fail", type="Flag", doc="Failure flag")
+        self.cubeKey = schema.addField(name+"_cube", type="D", doc="Cube of input catalog")
+
+    def burn(self, catalog):
+        for rec in catalog:
+            value = rec.get("start")
+            rec.set(self.cubeKey, value**3)
+
+    def fail(self, catalog, error=None):
+        for rec in catalog:
+            value = rec.set(self.failKey, True)
+
+@register("dependentAfterburner")
+class DependentAb(AB.AfterburnerPlugin):
+    '''
+    Afterburner plugin to test the runlevel resolution. This plugin takes in single records, reads a value
+    from a previous plugin computes a square root, and writes the results to the table.
+    '''
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_AFTERBURNER + 1
+
+    def __init__(self, config, name, schema, metadata):
+        AB.AfterburnerPlugin.__init__(self, config, name, schema, metadata)
+        self.failKey = schema.addField(name+"_fail", type="Flag", doc="Failure flag")
+        self.sqrtKey = schema.addField(name+"_sqrt", type="D", doc="Square root of singleRecord afterburner")
+
+    def burn(self, measRecord):
+        value = measRecord.get("singleRecordAfterburner_square")
+        measRecord.set(self.sqrtKey, value**0.5)
+
+    def fail(self, measRecord, error=None):
+        measRecord.set(self.failKey, True)
```

Comment:
Setting a `self.failKey` in the record seems to be a common pattern.  Can you support this in the base class?